packages/web-pkg/src/composables/driveResolver/useDriveResolver.ts

```diff
@@ -0,0 +1,85 @@
+import { useStore } from '../store'
+import { Store } from 'vuex'
+import { computed, Ref, ref, unref, watch } from '@vue/composition-api'
+import { buildSpace, SpaceResource } from 'web-client/src/helpers'
+import { useRouteQuery } from '../router'
+import { SHARE_JAIL_ID } from 'files/src/services/folder'
+import { useGraphClient } from 'web-client/src/composables'
+import { Resource } from 'web-client'
+import { useSpacesLoading } from './useSpacesLoading'
+
+interface DriveResolverOptions {
+  store?: Store<any>
+  driveAliasAndItem?: Ref<string>
+}
+
+export const useDriveResolver = (options: DriveResolverOptions = {}) => {
+  const store = options.store || useStore()
+  const { areSpacesLoading } = useSpacesLoading({ store })
+  const shareId = useRouteQuery('shareId')
+  const { graphClient } = useGraphClient({ store })
+  const spaces = computed(() => store.getters['runtime/spaces/spaces'])
+  const space = ref<SpaceResource>(null)
+  const item: Ref<string> = ref(null)
+  watch(
+    [options.driveAliasAndItem, areSpacesLoading],
+    ([driveAliasAndItem]) => {
+      if (!driveAliasAndItem) {
+        space.value = null
+        item.value = null
+        return
+      }
+      if (unref(space) && driveAliasAndItem.startsWith(unref(space).driveAlias)) {
+        item.value = driveAliasAndItem.slice(unref(space).driveAlias.length)
+        return
+      }
+      let matchingSpace = null
+      let path = null
+      if (driveAliasAndItem.startsWith('public/')) {
+        const [publicLinkToken, ...item] = driveAliasAndItem.split('/').slice(1)
+        matchingSpace = unref(spaces).find((s) => s.id === publicLinkToken)
+        path = item.join('/')
+      } else if (driveAliasAndItem.startsWith('share/')) {
+        const [shareName, ...item] = driveAliasAndItem.split('/').slice(1)
+        matchingSpace = buildSpace({
+          id: [SHARE_JAIL_ID, unref(shareId)].join('!'),
+          driveAlias: `share/${shareName}`,
+          driveType: 'share',
+          name: shareName,
+          shareId
+        })
```

Comment:
Add webDavPath?

Reply:
not needed, already handled internally in `buildSpace` :-)